### PR TITLE
Run active record tests on ruby 3.0+

### PR DESCRIPTION
**Pull request by tannalynn**
# Overview
This PR adds a new multiverse suite `active_record_pg` that runs the active record tests with postgres instead of mysql like the `active_record` suite uses. Mysql2 gem is difficult to get working on ruby 3.0+, so we moved to Postgres for all ruby 3.0+ active record tests. 

# Changes 
* Run jruby tests on Rails 5.0, 5.2, 6.0, 6.1 
* Add Postgres service to the multiverse CI actions
* Add activerecord-jdbcmysql-adapter and activerecord-jdbcsqlite-adapter to active record suite
* Refine Postgresql tests to support Rails 7.0+

# Related Github Issue
#902

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

**Comment by kaylareopelle** [comment]
**Note:** An earlier suggestion included reducing the duplication of files between the `active_record` and `active_record_pg` suites. Unfortunately, sharing a single test file raised errors with finding the database primary keys that I wasn't able to resolve. I'm happy to revisit this as a team if we'd like to continue efforts to reduce duplication.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -91,6 +91,18 @@ jobs:
     needs: build-ruby
     runs-on: ubuntu-18.04
     services:
+      postgres:
+        image: postgres:latest
+        env:
```
Comment (fallwith) [inline .github/workflows/ci.yml:96]: If we dropped these credentials ("postgres"/"postgres") and went with the defaults ("postgres"/"password") instead, then I believe you could just eliminate all related env vars from the config (both from the service config and the app configs), assuming they are what gets used when database.yml ends up with blank values.

File: test/multiverse/lib/multiverse/runner.rb [hunk 2]
```diff
@@ -107,6 +107,8 @@ def execute_suites(filter, opts)
     end
 
     if RUBY_VERSION >= '3.0'
+      # this suite uses mysql2 which has issues on ruby 3.0+
+      # a new suite, active_record_pg, has been added to run active record tests on ruby 3.0+
```
Comment (fallwith) [inline test/multiverse/lib/multiverse/runner.rb:111]: 👍

File: test/multiverse/suites/active_record/Envfile [hunk 3]
```diff
@@ -90,10 +90,12 @@ end
 # Only 3.2.22 fixes the above mentioned issue, so it's the only one 
 # surviving in our coverage of Rails 3.x series here.
 #
-if RUBY_VERSION <= '2.2.0'
+if RUBY_VERSION < '2.3'
+  mysql_vsn = '0.3.20'
```
Comment (fallwith) [inline test/multiverse/suites/active_record/Envfile:94]: I'm not sure if 0.3.21 gives us anything we can leverage, but it's the version I've been testing with (latest stable 0.3.x)

**Review by fallwith — APPROVED** [review]
👍

**Review by fallwith — APPROVED** [review]

**Pull request merged**